### ISSUE TITLE
React.js 19 nullable RefObject type is not supported

### ISSUE THREAD

**Issue by clacladev** [adobe/react-spectrum#7549]
### Provide a general summary of the issue here

In React 19 useRef and references in general have been refactored. Now they are stricter, they return null as potential value in their type.

`useRef<HTMLDivElement>(null)` will return:
- `RefObject<HTMLDivElement | null>` in React 19
- `RefObject<HTMLDivElement>` in React 18

A lot of React Aria hooks which accept a RefObject as input to create the component state, do not accept a nullable reference (`RefObject<HTMLDivElement | null>`).

For example
```
const calendarPopoverRef = useRef<HTMLDivElement>(null);
const { calendarProps } = useDatePicker<ZonedDateTime>(
    dateTimeInputProps,
    state,
    calendarPopoverRef, // <<< This will throw a Typescript error
);
```

### 🤔 Expected Behavior?

Hooks like `useDatePicker` should accept a `RefObject<Element | null>` type.
In general all packages should be checked against the TSC compiler.

### 😯 Current Behavior

It creates a TS error.

### 💁 Possible Solution

A temporary workaround is to add a cast to suppress the TS error. This is not a permanent solution though.
```
const calendarPopoverRef = useRef<HTMLDivElement>(null);
const { calendarProps } = useDatePicker<ZonedDateTime>(
    dateTimeInputProps,
    state,
    calendarPopoverRef as RefObject<Element>, // TODO: cleanup this type cast
);
```

### 🔦 Context

I am upgrading Nextjs 15.1 which also upgraded to stable React 19.

### 🖥️ Steps to Reproduce

- Upgrade you examples to have `react` and `@types/react` to v19
- Run `npm tsc`

### Version

"@react-aria/datepicker": "^3.11.0"

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

macOS

**Comment by devongovett** [comment]
I'm not seeing this issue. `useDatePicker` already does accept `RefObject<Element | null>`: https://github.com/adobe/react-spectrum/blob/f28f1f467623d65ced048a2dee7125c0abc43307/packages/%40react-aria/datepicker/src/useDatePicker.ts#L54

Here's a working example in the TS playground: [Example](https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbwK4GcCmAlNAzAvnbKCEOAcijQEMBjGUgbgChRJZFU0ARSmNABWDUA1migBlGDzT5CxMgAEKNGAFoUk3gBsAngHoAJlLCCRUUs3DR4ydN14DhomURKlFVWispRglA0ZNRBkZGaggAO3U4akpNNHDDKD4ISAA3USxsOABeOA5MgB4ACQAVAFkAGU5gVIBROJB4mAA+AApwpE1NAEomOPh1KRy82ykHUwkpVoRcXtCIqIRo2PjEviIwFDh8XI47fkCoVsY4U8RcABoTs8HeK7PluITvZLSMnCu5oA)

Is this still an issue?

**Comment by msandula12** [comment]
I'm experiencing this as well while trying to upgrade to React 19, with `useButton`, `useDialog`, and `useRangeCalendar`, among others.

```typescript
const ref = useRef<HTMLButtonElement>(null);
const { buttonProps } = useButton(props, ref);
```

> Argument of type 'RefObject<HTMLButtonElement | null>' is not assignable to parameter of type 'RefObject<Element>'.
>       Type 'HTMLButtonElement | null' is not assignable to type 'Element'.
>         Type 'null' is not assignable to type 'Element'.

**Comment by devongovett** [comment]
Please provide a reproduction.

**Comment by msandula12** [comment]
> Please provide a reproduction.
> 
> Make sure you're using the latest version of React Aria too. React 19 compatibility was added in [#6632](https://github.com/adobe/react-spectrum/pull/6632), released in https://react-spectrum.adobe.com/releases/2024-07-22.html (see list of versions at the bottom). You may need to check your lock file in case something got pinned to an older version, or you have multiple versions installed.

Yep. Wasn't on latest version of React Aria. Should've checked that first. Updating fixed the issue. Thanks!
